cpmpy/solvers/pysat.py

```diff
@@ -39,15 +39,69 @@
 """
 from .solver_interface import SolverInterface, SolverStatus, ExitStatus
 from ..exceptions import NotSupportedError
-from ..expressions.core import Expression, Comparison, Operator, BoolVal
+from ..expressions.core import Comparison, Operator, BoolVal
 from ..expressions.variables import _BoolVarImpl, NegBoolView, boolvar
 from ..expressions.globalconstraints import DirectConstraint
+from ..transformations.linearize import canonical_comparison
 from ..expressions.utils import is_int, flatlist
 from ..transformations.decompose_global import decompose_in_tree
 from ..transformations.get_variables import get_variables
 from ..transformations.flatten_model import flatten_constraint
 from ..transformations.normalize import toplevel_list, simplify_boolean
 from ..transformations.reification import only_implies, only_bv_reifies
+from ..expressions.utils import eval_comparison
+
+
+# TODO [?] where should we put this transformation function?
+# TODO [?] are zero coefficients removed in linearize? Probably better to do here than there (right now this is actually `only_non_negative_terms`).
```

Comment:
I don't remember why we made everything into wsums in the first place, but it feels a bit like this transformation is just undoing a previous transformation. Maybe we can just not make the wsum when it involves boolvars without coefficients?
In general I think we don't want transformations to be in the solver interface as well, so this one could be in the linearize file imo.

Reply:
I think linearize makes wsums because it's what a linear solver like gurobi expects. Note I'm not calling linearize in pysat's transform (exactly to avoid the wsum), I just call `canonical_comparison` to get all vars on the LHS. So I don't think this new "positive_coefficients" transform really undoes anything, but I could be wrong.

I agree I should move it to linearize anyway, even if PySAT is the only one that uses this transform for now.

Reply:
I guess see discussion below